File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetDeviceGroupPresentCapabilitiesKHX.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetDeviceGroupPresentCapabilitiesKHX_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pDeviceGroupPresentCapabilities, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkGetDeviceGroupPresentCapabilitiesKHX f = (PFN_vkGetDeviceGroupPresentCapabilitiesKHX)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkDeviceGroupPresentCapabilitiesKHX*)jniLongToPointer(pDeviceGroupPresentCapabilities))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkGetDeviceGroupPresentCapabilitiesKHX.java
package cz.mg.vulkan;

public class PFNvkGetDeviceGroupPresentCapabilitiesKHX extends VkFunctionPointer {
    public PFNvkGetDeviceGroupPresentCapabilitiesKHX() {
    }

    protected PFNvkGetDeviceGroupPresentCapabilitiesKHX(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetDeviceGroupPresentCapabilitiesKHX(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetDeviceGroupPresentCapabilitiesKHX(long value) {
        setValue(value);
    }

    public PFNvkGetDeviceGroupPresentCapabilitiesKHX(VkInstance instance) {
        super(instance, new VkString("vkGetDeviceGroupPresentCapabilitiesKHX"));
    }

    public void call(VkDevice device, VkDeviceGroupPresentCapabilitiesKHX pDeviceGroupPresentCapabilities, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pDeviceGroupPresentCapabilities != null ? pDeviceGroupPresentCapabilities.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void call(long vkaddress, long device, long pDeviceGroupPresentCapabilities, long rval);
}
